main.js

```diff
@@ -0,0 +1,128 @@
+const dropboxToken = core.getInput('token', { required: true });
+core.setSecret(dropboxToken);
+
+// 150MB per upload limit on the Dropbox API
+const MAX_UPLOAD_BYTES = 157286400;
+var filesToUpload = [];
+
+checkDropboxAuthentication();
+
+function checkDropboxAuthentication() {
+  const url = "https://api.dropboxapi.com/2/check/user";
+  const data = {
+    query: "Check Authentication"
+  }
+
+  axios({
+    url: url,
+    method: 'post',
+    headers: {
+      'Authorization' : 'Bearer ' + dropboxToken,
+      'Content-Type' : 'application/json'
+    },
+    data : JSON.stringify(data)
+  }).then(function (response) {
+    onAuthenticationSuccess(response);
+  }).catch(function (error) {
+    onAuthenticationFail(error);
+  });
+}
+
+function onAuthenticationSuccess(response) {
+  console.log('Auth response: ' + response.status);
+  console.log('Auth response: ' + response.statusText);
+
+  getDirFilesRecursive(srcPath);
+  uploadBuild();
+}
+
+function onAuthenticationFail(error) {
+  console.log(error);
+  core.setFailed(error);
+}
+
+function getDirFilesRecursive(dir) {
+  fs.readdirSync(dir).forEach(item => {
+    const fullPath = path.join(dir, item);
+    if (fs.lstatSync(fullPath).isDirectory()) {
+      getDirFilesRecursive(fullPath);
+    } else {
+      filesToUpload.push(fullPath);
+    }
+  });
+}
+
+function uploadBuild() {
+  uploadFile(filesToUpload[0], onUploadSuccess, onUploadFail);
+}
+
+function uploadFile(filePath, onSuccess, onFail) {
+  console.log("File: " + filePath);
+  const fileDstPath = filePath.replace(srcPath, dstPath);
+  console.log("Uploading to: " + fileDstPath);
+
+  const fileContent = fs.readFileSync(filePath);
+  const apiArgs = {
+    path: fileDstPath,
+    mute: true
+  }
+
+  const url = "https://content.dropboxapi.com/2/files/upload";
+  axios({
+    url: url,
+    method: 'post',
+    maxContentLength: MAX_UPLOAD_BYTES,
+    headers: {
+      'Authorization' : 'Bearer ' + dropboxToken,
+      'Content-Type' : 'application/octet-stream',
+      'Dropbox-API-Arg' : JSON.stringify(apiArgs)
+    },
+    data : fileContent
+  }).then(function (response) {
+    onSuccess(filePath, response);
+  }).catch(function (error) {
+    onFail(error);
+  });
+}
+
+function onUploadSuccess(localFilePath, response) {
+  console.log('Upload Successful: ' + localFilePath + '\n');
+
+  // As we only attempt to upload the next file on success of the previous, index should always be 0
+  let index = filesToUpload.indexOf(localFilePath);
+  if (index == -1) {
+    console.log("ERROR: Cannot pop " + localFilePath);
+    core.setFailed("Internal error");
+    return;
+  } else {
+    filesToUpload.splice(index, 1);
+  }
+
+  if (filesToUpload.length == 0) {
+    return;
+  }
+
+  uploadFile(filesToUpload[0], onUploadSuccess, onUploadFail);
+}
+
+function onUploadFail(error) {
+  if (error.response) {
+    // Try again if it's a rate limit error
+    if (error.response.headers['retry-after']) {
```

Comment:
Currently - nowhere!

I haven't been receiving this error since moving to upload files synchronously, so have left the handling of this error until it's actually a problem to avoid bloating the code-base.